shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/ContextManager.java

```diff
@@ -80,17 +81,21 @@
     
     private volatile InstanceContext instanceContext;
     
+    private volatile LockContext lockContext;
+    
     /**
      * Initialize context manager.
      *
      * @param metaDataContexts meta data contexts
      * @param transactionContexts transaction contexts
      * @param instanceContext instance context
+     * @param lockContext lock context
      */
-    public void init(final MetaDataContexts metaDataContexts, final TransactionContexts transactionContexts, final InstanceContext instanceContext) {
+    public void init(final MetaDataContexts metaDataContexts, final TransactionContexts transactionContexts, final InstanceContext instanceContext, final LockContext lockContext) {
         this.metaDataContexts = metaDataContexts;
         this.transactionContexts = transactionContexts;
         this.instanceContext = instanceContext;
+        this.lockContext = lockContext;
```

Comment:
Could `LockContext` be created inside `ContextManager` without passing parameter?

Reply:
Because `LockContext`  will synchronize the global lock state of the registry when `Proxy` starts.
Execute before register Online Instance in ContextManager can not be created inside now.
Will refactor later if possible